docs/csi-5to6-upgradepath.md

```diff
@@ -1,24 +1,48 @@
-3. If your cluster has more than 500 PVCs/Attachments, it is possible the time taking to construct database content exceed the restart threshold of the controller pods, reach out to PSO team if you run into this situation.
-
-## Instruction:
-1. prepare your `values.yaml` first for PSO 6 using the PSO 5.x `values.yaml`. They are formatted differently, so you won't be able to simply copy PSO 5.x `values.yaml` over. Schema validation will also help to make sure you comply with PSO 6 formats. 
-2. For upgrade field in PSO 6, it is by default set to `false`. PSO will not start the upgrade if the field is omitted or left to `false`. Set it to `true` to enable upgrade. The upgrade takes place during the installation. 
-3. Install PSO 6 following the instruction provided [here](../pure-pso/README.md). Monitor PSO pods and if all pods are up and running, the upgrade was successful. 
-
-Rolling back to PSO 5.x.
-1. Note that if upgrade is requested in `values.yaml` but the upgrade failed, PSO 6 will not start successfully. 
-2. If PSO 6 does not come up after 5 minutes, it is possible the upgrade failed, but other factors such as known network bug in certain k8s versions may also attribute to installation failures. To confirm the upgrade status, enable debug mode for PSO 6 in `values.yaml`, look into the pso-csi-container logs in pso-csi-container in the controller pod. If you need additional help, reach out to the Pure team for help. 
-3. At this point, you should be able to re-install PSO 5.x and not losing any existing resources on the cluster. First, check for k8s csi driver objects on the cluster and you will likely need to delete PSO 6 csi driver before you are able to reinstall PSO 5.x again. 
-run ```kubectl get csidriver``` to see a list of csi drivers installed on the cluster. 
+1. The upgrade process will **NOT** migrate snapshots or snapshots content from PSO 5.x to PSO 6.
+   * This restriction is enforced due to PSO 6 using the beta version of K8s snapshot API, whereas PSO 5.x used the alpha version of the API with backwards compatibility not being guranteed by the CSI standard. That means that a snapshot created in PSO 5.x cannot be used to restore a volume in PSO 6 after the upgrade.
+2. :warning: When installed PSO 6 it **MUST** use the same Pure namespace as PSO 5.x. 
+   * The `namespace` -> `pure` field in the PSO 5.x helm chart `values.yaml` must be **EXACTLY** the same as the `clusterID` field in PSO 6 `values.yaml` `namespace` field in PSO 5.x has been renamed to `clusterID` in PSO 6. (These parameters are used to prefix in underlying FlashArray volumes and FlashBlade shares used by PSO).
+   * :warning: Certain restrictions were imposed on PSO 6 `clusterID` and if your PSO 5.x `namespace` setting does not meet these restrictions, this upgrade tool will not work. Please reach out to Pure Storage Support for alternative options.
```

Comment:
Can we add a link to the clusterID restrictions?

Reply:
Not really documented anywhere, so I detailed them here now.